app/Http/Controllers/ConferencesController.php

```diff
@@ -46,28 +46,23 @@ public function index(Request $request)
 
         switch ($request->input('sort')) {
             case 'date':
+                $conferences = $conferences->sortBy->starts_at;
+                break;
+            case 'opening_next':
+                // Force CFPs with no CFP start date to the end
                 $conferences = $conferences->sortBy(function (Conference $model) {
-                    return $model->starts_at;
+                    return isset($model->cfp_starts_at) ? $model->cfp_starts_at : Carbon::now()->addCentury();
                 });
                 break;
-            case 'closing_next':
-                // Forces closed CFPs to the end. I feel dirty. Even dirtier with the 500 thing.
-                $conferences = $conferences
-                    ->sortBy(function (Conference $model) {
-                        if ($model->cfp_ends_at > Carbon::now()) {
-                            return $model->cfp_ends_at;
-                        } elseif ($model->cfp_ends_at === null) {
-                            return Carbon::now()->addYear(500);
-                        } else {
-                            return $model->cfp_ends_at->addYear(1000);
-                        }
-                    });
-                break;
             case 'alpha':
-                // Pass through
+                $conferences = $conferences->sortBy->title;
+                break;
+            case 'closing_next':
+                // pass through
             default:
+                // Forces closed CFPs to the end.
                 $conferences = $conferences->sortBy(function (Conference $model) {
-                    return strtolower($model->title);
+                    return isset($model->cfp_ends_at) ? $model->cfp_ends_at : Carbon::now()->addCentury();
```

Comment:
So, the original logic did two things:

1. Show those which close in the future, sorted by how soon
2. Show those which have no CFP close date
3. Show those which have already closed.

This current re-work of it gets rid of #3 (in the order) and now, if a CFP closed 4 weeks ago, it'll be higher in the order than one which closes in a week, which defeats the purpose, even if it is more logical.

What do you think about that?